core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,322 @@
+    editorContextService, alertsService, INTERACTION_SPECS,
+    stateSolutionService, explorationStatesService,
+    oppiaExplorationHtmlFormatterService, stateInteractionIdService,
+    stateHintsService, UrlInterpolationService, SolutionObjectFactory,
+    responsesService, AnswerClassificationService,
+    explorationContextService, angularNameService, oppiaHtmlEscaper) {
+    $scope.editorContextService = editorContextService;
+    $scope.stateSolutionService = stateSolutionService;
+    $scope.isActive = false;
+    $scope.answerIsExclusive = false;
+    $scope.answerCheckPassed = false;
+    $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+    $rootScope.interactionHtml = '';
+
+    $scope.stateHintsService = stateHintsService;
+    $scope.stateSolutionService = stateSolutionService;
+    $rootScope.correctAnswer = null;
+
+    $scope.supportedInteractions = [
+      'InteractiveMap',
+      'MusicNotesInput',
+      'GraphInput',
+      'SetInput',
+      'MathExpressionInput',
+      'MultipleChoiceInput',
+      'ImageClickInput',
+      'ItemSelectionInput'
+    ];
+
+    $scope.unsupportedInteractionObjectTypes = {
+      CodeRepl: 'CodeString',
+      PencilCodeEditor: 'CodeString',
+      NumericInput: 'Real',
+      TextInput: 'NormalizedString',
+      LogicProof: 'LogicQuestion'
+    };
+
+    $rootScope.submitAnswer = function (answer, rules) {
+      $rootScope.correctAnswer = answer;
+    };
+
+    $scope.toggleIsActive = function() {
+      $scope.isActive = !$scope.isActive;
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $rootScope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          'currentInteractionHtml'));
+
+      $scope.solutionIsSpecified = (
+        JSON.stringify(stateSolutionService.savedMemento) ? true : false);
+      $scope.isActive = false;
+      var interactionId = stateInteractionIdService.savedMemento;
+
+      $scope.INTERACTION_SPECS = INTERACTION_SPECS[interactionId];
+
+      if ($scope.supportedInteractions.indexOf(interactionId) === -1) {
+        $rootScope.interactionHtml = null;
+        $rootScope.objectType = (
+          $scope.unsupportedInteractionObjectTypes[interactionId]);
+      } else {
+        $rootScope.objectType = null;
+      }
+    });
+
+    $scope.getSolutionSummary = function() {
+      var solution = stateSolutionService.displayed;
+      var isExclusiveAnswer = (
+        solution.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = '';
+      var interactionId = stateInteractionIdService.savedMemento;
+      if (interactionId === 'GraphInput') {
```

Comment:
See if you can find opportunities to extract some of the logic into services, in order to increase testability. This controller is too big.